lib/_emerge/depgraph.py

```diff
@@ -1234,7 +1234,11 @@ def _show_ignored_binaries(self):
     def _show_ignored_binaries_respect_use(self, respect_use):
         seen = {}
         messages = []
-        merging = {pkg.cpv for pkg in self._dynamic_config._displayed_list}
+        merging = {
+            pkg.cpv
```

Comment:
Since this list can contain packages from multiple roots, it would make sense to make this a tuple that includes pkg.root (EROOT) as in 935d1432b2ea42fc0a49b9e3368a1766a673e1a9.

Reply:
Ah, good catch. I've added a test for the original bug so now I'll look at this.